File: examples/tracing-masking/langfuse-mask.mjs
/**
 * The live Langfuse JS integration. `await initialize()` must resolve
 * before this module's `scanAndRedact` is used
 * (`packages/javascript/src/index.ts`); `createMaskSecrets` enforces that
 * order so a caller can't forget it. Not imported by
 * `mask-secrets.test.mjs` — like `examples/safe-integration/server.mjs`,
 * this thin file is exercised against the real built package, not unit
 * tested with a fake.
 *
 * Langfuse JS's `mask` option receives `{ data }`, where `data` is the
 * stringified JSON of the attribute's value, and must return the masked
 * string (https://langfuse.com/docs/observability/features/masking,
 * confirmed against the current docs while resolving issue #326):
 *
 * ```js
 * import { Langfuse } from "langfuse";
 * import { createMaskSecrets } from "./langfuse-mask.mjs";
 *
 * const maskSecrets = await createMaskSecrets();
 * const langfuse = new Langfuse({ mask: ({ data }) => maskSecrets(data) });
 * ```
 *
 * `maskSecrets` also accepts a parsed object/array directly (not only a
 * JSON string), so the same function works for the generic
 * "mask this payload before tracing" case the issue describes, not only
 * Langfuse's stringified form.
 */

import { initialize, scanAndRedact } from "@redact-secret/core";

import { maskSecretsWith } from "./mask-secrets.mjs";

/**
 * Awaits `initialize()` once, then returns `maskSecrets(data)`: a
 * single-argument function matching the issue's proposed signature and
 * ready to wrap for Langfuse's `mask: ({ data }) => maskSecrets(data)`.
 */
export async function createMaskSecrets(options = {}) {
  await initialize();
  return (data) => maskSecretsWith(scanAndRedact, data, options);
}

File: examples/tracing-masking/python/langfuse_mask.py
"""The live Langfuse Python integration.

Langfuse Python's legacy `mask` hook receives the actual attribute value
(not a stringified form, unlike the JS SDK) and must return the masked
value:

    def masking_function(*, data: Any, **kwargs: Any) -> Any: ...

(https://langfuse.com/docs/observability/features/masking, confirmed
against the current docs while resolving issue #326.) `mask_secrets`
below matches that signature exactly, so it can be passed directly:

    from langfuse import Langfuse
    from langfuse_mask import mask_secrets

    langfuse = Langfuse(mask=mask_secrets)

Not imported by `test_mask_secrets.py` -- like
`examples/safe-integration`'s live wrapper files on the JS side, this thin
module is exercised against the real built package, not unit tested with
a fake. There is no init step for the Python bindings, unlike the JS
package's `await initialize()`.
"""

from __future__ import annotations

from typing import Any

import redact_secret

from mask_secrets import mask_secrets_with

__all__ = ["mask_secrets"]


def mask_secrets(*, data: Any, **_kwargs: Any) -> Any:
    return mask_secrets_with(redact_secret.scan_and_redact, data)
